src/utils/nodeTypeProcessing.ts

```diff
@@ -281,29 +285,39 @@ export function isStorageSpecificType(
   ) {
     visitedStructs.push(type.definition.id);
     return type.definition.vMembers.some((m) =>
-      isStorageSpecificType(safeGetNodeType(m, ast.compilerVersion), ast, visitedStructs),
+      isStorageSpecificType(safeGetNodeType(m, ast.inference), ast, visitedStructs),
     );
   }
   return false;
 }
 
-export function safeGetNodeType(node: Expression | VariableDeclaration, version: string): TypeNode {
+export function safeGetNodeType(
+  node: Expression | VariableDeclaration,
+  inference: InferType,
+): TypeNode {
   getContainingSourceUnit(node);
-  return getNodeType(node, version);
+  // if (node instanceof VariableDeclaration) {
+  //   return inference.variableDeclarationToTypeNode(node);
+  // }
+  // if (node instanceof TypeName) {
+  //   return inference.typeNameToTypeNode(node)
+  // }
+  // return inference.typeOf(node);
```

Comment:
This logic?

Reply:
It's the logic to run inferType instead of typestring parser. I left that so it would be easy to just uncomment that and run tests to check how the new implementation works with InferType as the transition between the two will take some time